File: pinyougou-manager-web/src/main/webapp/js/pinyougou/controller/sellerController.js
app.controller("sellerController",function($scope,$controller,sellerService){

    /*继承父模块*/
    $controller("baseController",{$scope:$scope});

    $scope.findAll=function(){
        sellerService.findAll().success(function (responseData) {
            $scope.sellerList=responseData;
        });
    }

    $scope.searchEntity={};
    /*分页查询*/
    $scope.findPage=function(page,rows){
        sellerService.findPage(page,rows,$scope.searchEntity).success(function(responseData){
            $scope.sellerList=responseData.rows;
            //更新总记录数
            $scope.paginationConf.totalItems=responseData.total;
        });
    }

    /*保存*/
    $scope.save=function(){
        sellerService.save($scope.seller).success(function(responseData){
            if (responseData.success){
                alert(responseData.message);
                $scope.reloadList();
            }else{
                alert(responseData.message);
            }
        });
    };

    /*查询单个品牌信息*/
    $scope.findOne=function(id){
        sellerService.findOne(id).success(function(responseData){
            $scope.seller=responseData;
        });
    };

    /*批量删除*/
    $scope.deleteBatch=function(){
        sellerService.deleteBatch($scope.selectIds).success(function(responseData){
            if (responseData.success){
                alert(responseData.message);
                $scope.reloadList();
                $scope.selectIds=[];
            }else{
                alert(responseData.message);
            }
        });
    };

    /*更新商家状态*/
    $scope.updateStatus=function(sellerId,status){
        sellerService.updateStatus(sellerId,status).success(function(responseData){
            if (responseData.success){
                alert(responseData.message);
                $scope.reloadList();
            }else{
                alert(responseData.message);
            }
        });
    };

    /*定义一个数组来存储商家状态*/
    $scope.status=['未审核','审核通过','审核未通过','关闭'];

});

File: pinyougou-manager-web/src/main/java/com/rogercw/pinyougou/web/controller/SellerController.java
package com.rogercw.pinyougou.web.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.rogercw.page.PageResult;
import com.rogercw.pinyougou.pojo.Seller;
import com.rogercw.pinyougou.service.SellerService;
import com.rogercw.result.ResponseMsg;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Date;
import java.util.List;

/**
 * @Author: rogercw
 * @Date: 2018/8/29 23:15
 * @Version 1.0
 */
@RestController
@RequestMapping("/seller")
public class SellerController {

    @Reference
    private SellerService sellerService;

    @RequestMapping("/findAll")
    public List<Seller> findAll(){
        return sellerService.findAll();
    }

    @RequestMapping("/findPage")
    public PageResult findPage(int page,int rows,@RequestBody Seller seller){
        return sellerService.findPage(seller,page, rows);
    }

    @RequestMapping("/findOne")
    public Seller findOne(String id){
        return sellerService.findOne(id);
    }

    @RequestMapping("/delete")
    public ResponseMsg delete(String[] ids){
        try {
            sellerService.delete(ids);
            return new ResponseMsg(true,"删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new ResponseMsg(false,"删除失败");
        }
    }


    @RequestMapping("/updateStatus")
    public ResponseMsg updateStatus(String sellerId,String status){
        try {
            sellerService.updateStatus(sellerId,status);
            return new ResponseMsg(true,"更新状态成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new ResponseMsg(false,"更新状态失败");
        }
    }
}
